pkg/autodiscovery/listeners/kube_services.go

```diff
@@ -204,6 +203,19 @@ func (l *KubeServiceListener) createService(ksvc *v1.Service, firstRun bool) {
 	if ksvc == nil {
 		return
 	}
+
+	l.m.Lock()
+	l.endpointsAnnotated[fmt.Sprintf("%s/%s", ksvc.Namespace, ksvc.Name)] = isEndpointAnnotated(ksvc)
+	endpoints, err := l.endpointsInformer.Lister().List(labels.Everything())
```

Comment:
You only need to Get the corresponding endpoints object (same namespace, same name)